src/core/stream/representation/representation_stream.ts

```diff
@@ -377,13 +382,96 @@ export default function RepresentationStream<T>({
 
       case "parsed-segment":
         return initSegmentState.segmentData$
-          .pipe(mergeMap((initSegmentData) =>
-            pushMediaSegment({ clock$,
-                               content,
-                               initSegmentData,
-                               parsedSegment: evt.value,
-                               segment: evt.segment,
-                               segmentBuffer })));
+          .pipe(mergeMap((initSegmentData) => {
+            if (evt.value.inbandEvents !== undefined) {
```

Comment:
Can we move some of this to its own function.
I;m having troubles following where that `if` ends!